build.assets/tooling/cmd/notarize-apple-binaries/hashicorpLoggerAdapter.go

```diff
@@ -0,0 +1,228 @@
+func (hclla HCLogLogrusAdapter) Name() string {
+	return hclla.name
+}
+
+// Create a logger that will prepend the name string on the front of all messages.
+// If the logger already has a name, the new value will be appended to the current
+// name. That way, a major subsystem can use this to decorate all it's own logs
+// without losing context.
+func (hclla HCLogLogrusAdapter) Named(name string) hclog.Logger {
+	if hclla.name != "" {
+		name = fmt.Sprintf("%s.%s", name, hclla.name)
+	}
+
+	return hclla.ResetNamed(name)
+}
+
+// Create a logger that will prepend the name string on the front of all messages.
+// This sets the name of the logger to the value directly, unlike Named which honor
+// the current name as well.
+func (HCLogLogrusAdapter) ResetNamed(name string) hclog.Logger {
+	newLogger := NewHCLogLogrusAdapter()
+	newLogger.name = name
+
+	return newLogger
+}
+
+// Updates the level. This should affect all related loggers as well,
+// unless they were created with IndependentLevels. If an
+// implementation cannot update the level on the fly, it should no-op.
+func (hclla HCLogLogrusAdapter) SetLevel(level hclog.Level) {
+	hclla.logger.SetLevel(hclogLevelToLogrusLevel(level))
+}
+
+// Returns the current level
+func (hclla HCLogLogrusAdapter) GetLevel() hclog.Level {
+	switch hclla.logger.GetLevel() {
+	case logrus.FatalLevel:
+	case logrus.PanicLevel:
+	case logrus.ErrorLevel:
+		return hclog.Error
+	case logrus.WarnLevel:
+		return hclog.Warn
+	case logrus.InfoLevel:
+		return hclog.Info
+	case logrus.DebugLevel:
+		return hclog.Debug
+	case logrus.TraceLevel:
+		return hclog.Trace
+	}
+
+	return hclog.NoLevel
+}
+
+// Return a value that conforms to the stdlib log.Logger interface
+// Options are ignored as it's not currently worth the time to implement them
+func (hclla HCLogLogrusAdapter) StandardLogger(options *hclog.StandardLoggerOptions) *log.Logger {
+	return log.New(hclla.StandardWriter(options), "", 0)
+}
+
+// Return a value that conforms to io.Writer, which can be passed into log.SetOutput()
+// Options are ignored as it's not currently worth the time to implement them
+func (hclla HCLogLogrusAdapter) StandardWriter(_ *hclog.StandardLoggerOptions) io.Writer {
+	logrusOut := hclla.logger.Out
+	if logrusOut != nil {
+		return logrusOut
+	}
+
+	return os.Stderr
+}
+
+func hclogLevelToLogrusLevel(level hclog.Level) logrus.Level {
+	// The docs for hclog also list a "Off" level, but it's not defined in the version of hclog used by gon.
+	switch level {
+	case hclog.Error:
+		return logrus.ErrorLevel
+	case hclog.Warn:
+		return logrus.WarnLevel
+	case hclog.Info:
+		return logrus.InfoLevel
+	case hclog.Debug:
+		return logrus.DebugLevel
+	case hclog.Trace:
+		return logrus.TraceLevel
+	case hclog.NoLevel:
+		return logrus.InfoLevel
+	}
+
+	return logrus.InfoLevel
+}
+
+func (hclla HCLogLogrusAdapter) createImpliedArgsString() string {
+	keyPairCount := len(hclla.impliedArgs) / 2
+	keyPairString := ""
```

Comment:
Would using a [strings.Builder](https://pkg.go.dev/strings#Builder) be useful here?

Reply:
**Thank you**! I was looking for something like this earlier, but didn't know the Go name for it. Updated